src/apps/engine/CMakeLists.txt

```diff
@@ -1,8 +1,13 @@
+include(${CMAKE_BINARY_DIR}/conanbuildinfo_multi.cmake)
+
 STORM_SETUP(
     TARGET_NAME engine
     TYPE executable
-    DEPENDENCIES     
-	
+    DEPENDENCIES
+    
+    # this is something I really don't like but it seems there's not better way to ensure mimalloc is linked before c-runtime
+    $<IF:$<CONFIG:Release>,${CONAN_LIB_DIRS_MIMALLOC_RELEASE}/mimalloc.lib,${CONAN_LIB_DIRS_MIMALLOC_DEBUG}/mimalloc-debug.lib>
```

Comment:
Can you explain why you need to link mimalloc before c-runtime?
And if you really need it - try to avoid direct filenames: for example bebug linux lib called `libmimalloc-debug.so`

Reply:
The idea is to override both C++ and C std memory handling. forC++ there is a well-defined way to override all variations of new/delete operators. 
However, for lower-level C alloc stuff it's not that simple. On windows, the allocator can only be injected at runtime.
This creates a constraint for linker/loader order. mimalloc-redirect shall be loaded strictly before C runtime (for modern windows it is ucrtbase/ucrtbased).

Regarding hardcoded names. I have no idea how we can deduce the names. I tried a few techniques and they do not work. We are heavily constrained by Conan API here.
One idea I have in mind - on Linux I think it can be statically linked. Therefore we can probably configure it at conanfile level depending on the platform. And disable the above code on Linux. But this is not something I can do myself right now. So feel free to either try it yourself or disable this code for your fork for now

Reply:
good catch

Reply:
I tested few vars like `${CONAN_LIBS_MIMALLOC_RELEASE}` and `${CONAN_PKG_LIBS_MIMALLOC_RELEASE}`, but they don't fit. So I can offer simple solution, which works fine on Windows and Linux
```
$ git diff
diff --git a/src/apps/engine/CMakeLists.txt b/src/apps/engine/CMakeLists.txt
index 1008563e..0a94b106 100644
--- a/src/apps/engine/CMakeLists.txt
+++ b/src/apps/engine/CMakeLists.txt
@@ -1,12 +1,18 @@
 include(${CMAKE_BINARY_DIR}/conanbuildinfo_multi.cmake)

+if (WIN32)
+set(MIMALLOC_DEP "$<IF:$<CONFIG:Release>,${CONAN_LIB_DIRS_MIMALLOC_RELEASE}/mimalloc.lib,${CONAN_LIB_DIRS_MIMALLOC_DEBUG}/mimalloc-debug.lib>")
+else()
+set(MIMALLOC_DEP "$<IF:$<CONFIG:Release>,${CONAN_LIB_DIRS_MIMALLOC_RELEASE}/libmimalloc.so,${CONAN_LIB_DIRS_MIMALLOC_DEBUG}/libmimalloc-debug.so>")
+endif()
+
 STORM_SETUP(
     TARGET_NAME engine
     TYPE executable
     DEPENDENCIES

     # this is something I really don't like but it seems there's not better way to ensure mimalloc is linked before c-runtime
-    $<IF:$<CONFIG:Release>,${CONAN_LIB_DIRS_MIMALLOC_RELEASE}/mimalloc.lib,${CONAN_LIB_DIRS_MIMALLOC_DEBUG}/mimalloc-debug.lib>
+    ${MIMALLOC_DEP}

     # internal
     common
```

Reply:
I hope we'll find a better way... but let it be for now :)